File: library/src/main/java/com/adedom/library/Pathiphon.java
package com.adedom.library;

import android.os.StrictMode;

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;

public class Pathiphon {

    private static final String IP = "192.168.43.22";
    private static final String DB_NAME = "the_egg_game";
    private static final String USERNAME = "root";
    private static final String PASSWORD = "abc456";
    public static final String BASE_URL = "http://" + IP + "/the-egg-game/";

    public static Connection connection() {
        return connection(IP, DB_NAME, USERNAME, PASSWORD);
    }

    @Deprecated
    public static Connection connection(String ip, String dbName, String username, String password) {
        StrictMode.setThreadPolicy(new StrictMode.ThreadPolicy.Builder().permitAll().build());

        try {
            Class.forName("com.mysql.jdbc.Driver");
            String url = "jdbc:mysql://" + ip + "/" + dbName + "?useUnicode=true&characterEncoding=utf-8";
            return DriverManager.getConnection(url, username, password);
        } catch (ClassNotFoundException e) {
            e.printStackTrace();
        } catch (SQLException e) {
            e.printStackTrace();
        }
        return null;
    }

    @Deprecated
    public static Execute connection(Connection connection) {
        return new Execute(connection);
    }

    @Deprecated
    public static Call with(Connection connection) {
        return new Call(connection);
    }

    public static ExecuteCommit execute(String sql) {
        return new ExecuteCommit(connection(), sql);
    }

    public static CallCommit call(String storedProcedureName) {
        return new CallCommit(connection(), storedProcedureName);
    }
}

File: utility/src/main/java/com/adedom/utility/Pathiphon.kt
package com.adedom.utility

object Pathiphon {
    private val HOST = "192.168.43.22"
    val BASE_URL = "http://$HOST/the-egg-game/api/"

//    val BASE_URL = "http://pathiphon.hostingerapp.com/api/"


//    companion object {
//        private val USERNAME = "root"
//        private val PASSWORD = "abc456"
//        private val DB_NAME = "the_egg_game"
//        fun connection(): Connection? {
//            return connection(HOST, USERNAME, PASSWORD, DB_NAME)
//        }
//
//        fun connection(
//            ip: String,
//            username: String,
//            password: String,
//            dbName: String
//        ): Connection? {
//            StrictMode.setThreadPolicy(StrictMode.ThreadPolicy.Builder().permitAll().build())
//
//            try {
//                Class.forName("com.mysql.jdbc.Driver")
//                val url = "jdbc:mysql://$ip/$dbName?useUnicode=true&characterEncoding=utf-8"
//                return DriverManager.getConnection(url, username, password)
//            } catch (e: ClassNotFoundException) {
//                e.printStackTrace()
//            } catch (e: SQLException) {
//                e.printStackTrace()
//            }
//
//            return null
//        }
//
//        @Deprecated("")
//        fun connection(connection: Connection): Execute {
//            return Execute(connection)
//        }
//
//        @Deprecated("")
//        fun with(connection: Connection): Call {
//            return Call(connection)
//        }
//
//        fun execute(sql: String): ExecuteCommit {
//            return ExecuteCommit(connection(), sql)
//        }
//
//        fun call(storedProcedureName: String): CallCommit {
//            return CallCommit(connection(), storedProcedureName)
//        }
//    }
}